provider/core/src/datagen/payload.rs

```diff
@@ -143,3 +169,28 @@ pub struct ExportMarker {}
 impl DataMarker for ExportMarker {
     type Yokeable = ExportBox;
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::hello_world::*;
+
+    #[test]
+    fn test_compare_with_dyn() {
+        let payload1: DataPayload<HelloWorldV1Marker> = DataPayload::from_owned(HelloWorldV1 {
```

Comment:
What's the plan for comparing `DataPayload<ExportMarker>`, i.e. the upcasted one? Are you planning to do that in a separate PR?